### PR TITLE
Add a new go-dnsclient for scale tests.

### PR DESCRIPTION
**Pull request by prameshj**
This client runs DNS lookups for the specified duration and pushes up metrics.

**What type of PR is this?**
/kind feature

This PR adds a new DNS client to be used in scale testing. This new client pushes up prometheus metrics.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: dns/dnsperfgo/main.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+				// Wait for the test to run for testDuration seconds before firing. Without this reset, the test will only run for (testDuration - idleDuration) seconds.
+				ticker.Reset(config.testDuration + config.idleDuration)
+				time.Sleep(config.idleDuration)
+				log.Print("Retarting DNS lookups.")
+			default:
+				break
+			}
+			// Use nsLookup command rather than net.LookupHost because nslookup sends A and AAAA lookups in parallel(with the same source port) in Alpine base image.
+			// Go program sends them with different source ports. The same source port behavior will trigger the DNS race conditions described in https://www.weave.works/blog/racy-conntrack-and-dns-lookup-timeouts.
+			err, timedOut := runQuery(h, config.queryTimeout, nsLookup)
```
Comment (dlapcevic) [inline dns/dnsperfgo/main.go:99]: Does waiting for this query to finish still mean that we run the expected number of queries indicated by the qps?

Comment (prameshj) [reply dns/dnsperfgo/main.go]: You sre right.. This wait can cause the actual qps to be lower. Addressed by moving it to a goroutine and verifying the timeout case in a local pod.

File: dns/dnsperfgo/main.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+				c.logResults()
+				log.Print("Exiting.")
+				c.metricsServer.Shutdown(context.TODO())
+				return
+			case <-ticker.C:
+				c.logResults()
+				// Wait for the test to run for testDuration seconds before firing. Without this reset, the test will only run for (testDuration - idleDuration) seconds.
+				ticker.Reset(c.config.testDuration + c.config.idleDuration)
+				time.Sleep(c.config.idleDuration)
+				log.Print("Retarting DNS lookups.")
```
Comment (dlapcevic) [inline dns/dnsperfgo/main.go:103]: nit: Restarting

File: dns/dnsperfgo/main.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	var ip string
+	var index int
+	ips := make([]string, 0)
+	outputStrings := strings.Split(string(out), "\n")
+	for _, s := range outputStrings {
+		if strings.Contains(s, "Address ") {
+			// String will be of the form - "Address 1: 10.99.48.1 kubernetes.default.svc.cluster.local" or "Address 1: 40.76.4.15" on Alpine/BusyBox. Read in ip and host as a single variable.
+			_, err := fmt.Sscanf(s, "Address %d: %s", &index, &ip)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to scan string %q, err - %w", s)
```
Comment (dlapcevic) [inline dns/dnsperfgo/main.go:190]: nit: Start error message with a non-capital letter.

**Comment by prameshj** [comment]
verify-lint test failed because linter is using Golang < 1.15.
verify-test is failing because prow is using Golang < 1.15 for building.
benchmark-kube-dns is failing because jsonify/main.go is not using go modules:

```
INFO 11-19 07:14:29 runner.py:426] Client teardown complete
jsonify/main.go:29:2: no required module provides package github.com/golang/glog: go.mod file not found in current directory or any parent directory; see 'go help modules'
jsonify/main.go:30:2: no required module provides package gopkg.in/yaml.v2: go.mod file not found in current directory or any parent directory; see 'go help modules'
jsonify/main.go:31:2: no required module provides package k8s.io/kubernetes/test/e2e/perftype: go.mod file not found in current directory or any parent directory; see 'go help modules'
```

**Pull request merged**